src/builder/esbuild.ts

```diff
@@ -34,110 +34,130 @@ export class EsbuildBuilder extends BaseBuilder<BuildOptions> {
+        watch: {
+          onRebuild: (error: unknown) => {
+            if (error) {
+              logger.error('Refresh', this.env.toLowerCase(), 'failed', error)
+            } else {
+              logger.log('Refresh', this.env.toLowerCase())
               start()
+            }
+          },
+        },
+      })) as BuildResult
 
-              await watcher.close()
-              this.dev(start)
-            },
-            { wait: 200 },
-          ),
-        )
-      })
+      start()
     } else if (isRenderer(this.config)) {
       if (typeof this.config.fileConfig.html === 'undefined') {
         logger.end('Cannot use esbuild in renderer without specifying a html file in `rendererConfig.html`')
+        return
       }
 
-      const srcDir = path.resolve(process.cwd(), path.dirname(this.config.fileConfig.src))
+      const htmlPath = path.resolve(process.cwd(), this.config.fileConfig.html)
 
-      esbuild
-        .serve(
-          {
-            host: 'localhost',
-            port: 9081,
-          },
-          this.config.config,
+      const html = (await fs.readFile(htmlPath))
+        .toString()
+        .replace(
+          '</body>',
+          '<script>new EventSource("/events").onmessage = () => window.location.reload();</script></body>',
         )
-        .then(async (builder) => {
-          if (typeof this.config.fileConfig.html === 'undefined') {
-            logger.end('Cannot use esbuild in renderer without specifying a html file in `rendererConfig.html`')
-            return
-          }
-
-          const livereloadPort = 35729
-          const htmlPath = path.resolve(process.cwd(), this.config.fileConfig.html)
-          const html = (await fs.readFile(htmlPath))
-            .toString()
-            .replace('</body>', `<script src="/livereload.js?snipver=1"></script></body>`)
-
-          const proxy = httpProxy.createProxy({ target: 'http://localhost:9081' })
-          const proxyLr = httpProxy.createProxy({ target: `http://localhost:${livereloadPort}` })
-          const lrServer = livereload.createServer({ port: livereloadPort })
-
-          const handler = connect()
-
-          handler.use(compression() as never)
-          handler.use((req, res) => {
-            if (req.url === '/' || req.url === '') {
-              res.setHeader('Content-Type', 'text/html')
-              res.setHeader('Cross-Origin-Embedder-Policy', 'require-corp')
-              res.setHeader('Cross-Origin-Opener-Policy', 'same-origin')
-              res.writeHead(200)
-              res.end(html)
-            } else if (req.url?.includes('livereload.js')) {
-              proxyLr.web(req, res)
-            } else {
-              proxy.web(req, res)
-            }
-          })
 
-          const server = createServer(handler)
+      const handler = connect()
 
-          server.on('upgrade', (req, socket, head) => {
-            proxyLr.ws(req, socket, head)
-          })
+      // handler.use(compression() as never)
```

Comment:
`compression` seems to break SSE. I'm assuming there's a fix but I haven't looked into it yet.

Reply:
https://github.com/expressjs/compression#server-sent-events